index.bs

```diff
@@ -3100,22 +3100,31 @@ The [=remote end steps=] with |command parameters| are:
 
 1. Let |device pixel ratio| be the |command parameters|["<code>devicePixelRatio</code>"] field.
 
-1. If |viewport| is not null, and the implementation is unable to set the width and height dimensions
-   of the [=layout viewport=] to |viewport|["width"] and |viewport|["height"], respectively, or if 
-   |device pixel ratio| is not null and the implementation is unable to set the size of the CSS pixel 
-   of the [=layout viewport=] to |device pixel ratio| device pixels, or if the implementation is unable
-   to adjust the [=layout viewport=] parameters for any other reason, return [=error=] 
-   with [=error code=] [=unsupported operation=].
+1. If the implementation is unable to set the [=layout viewport=] for any
```

Comment:
I think we want to be specific here about the operations related to the viewport. Setting the viewport is not precise enough.

Reply:
we also only need to throw an error if the attribute is actually specified. Let's revert to the previous definition?

Reply:
How about "If the implementation is unable to perform any operations defined in this algorithm for any reason, ..."?

Reply:
I've phrased it to be the third part of the previous definition. I think being so specific is a bit redundant since the only two operations are setting dpr and setting viewport.

Reply:
I've separated the previous warning and moved them into their corresponding locations.